pkg/exchange/okex/kline_stream.go

```diff
@@ -84,11 +90,22 @@ func (s *KLineStream) dispatchEvent(e interface{}) {
 
 func (s *KLineStream) Unsubscribe() {
 	// errors are handled in the syncSubscriptions, so they are skipped here.
+	instType := s.getInstrumentType()
 	if len(s.StandardStream.Subscriptions) != 0 {
-		_ = syncSubscriptions(s.StandardStream.Conn, s.StandardStream.Subscriptions, WsEventTypeUnsubscribe)
+		_ = syncSubscriptions(s.StandardStream.Conn, s.StandardStream.Subscriptions, WsEventTypeUnsubscribe, instType)
 	}
 	s.Resubscribe(func(old []types.Subscription) (new []types.Subscription, err error) {
 		// clear the subscriptions
 		return []types.Subscription{}, nil
 	})
 }
+
+func (s *KLineStream) getInstrumentType() okexapi.InstrumentType {
+	if s.exchange.IsMargin {
```

Comment:
you can call s.exchange.getInstrumentType, since it's in the same package, calling its private method is doable.

Reply:
done